tests/flow/test_concurrent_query.py

```diff
@@ -10,64 +8,74 @@
 GRAPH_ID = "G"                      # Graph identifier.
 CLIENT_COUNT = 16                   # Number of concurrent connections.
 graphs = None                       # One graph object per client.
-assertions = [True] * CLIENT_COUNT  # Each thread places its verdict at position threadID.
-exceptions = [None] * CLIENT_COUNT  # Each thread which fails sets its exception content ar position threadID.
 people = ["Roi", "Alon", "Ailon", "Boaz", "Tal", "Omri", "Ori"]
 
 def query_aggregate(graph, query, threadID):
-    global assertions
-    assertions[threadID] = True
-
     for i in range(10):
         actual_result = graph.query(query)
         person_count = actual_result.result_set[0][0]
         if person_count != len(people):
-            assertions[threadID] = False
-            break
+            return False
+    
+    return True
 
 def query_neighbors(graph, query, threadID):
-    global assertions
-    assertions[threadID] = True
-
     # Fully connected graph + header row.
     expected_resultset_size = len(people) * (len(people)-1)
 
     for i in range(10):
         actual_result = graph.query(query)
         if len(actual_result.result_set) is not expected_resultset_size:
-            assertions[threadID] = False
-            break
+            return False
+    
+    return True
 
 def query_write(graph, query, threadID):
-    global assertions
-    assertions[threadID] = True
-    
     for i in range(10):
         actual_result = graph.query(query)        
         if actual_result.nodes_created != 1 or actual_result.properties_set != 1:
-            assertions[threadID] = False
-            break
+            return False
+
+    return True
 
 def thread_run_query(graph, query, threadID):
-    global assertions
     try:
-        assertions[threadID] = graph.query(query)
+        return graph.query(query)
     except ResponseError as e:
-        exceptions[threadID] = str(e)
+        return str(e)
 
 def delete_graph(graph, threadID):
-    global assertions
-    assertions[threadID] = True
-
     # Try to delete graph.
     try:
         graph.delete()
+        return True
     except:
         # Graph deletion failed.
-        assertions[threadID] = False
+        return False
+
+def run_concurrent(env, queries, f):
+    indexes = range(CLIENT_COUNT)
+    pool = Pool(nodes=CLIENT_COUNT)
+
+    # invoke queries
+    m = pool.amap(f, graphs, queries, indexes)
```

Comment:
it is unclear that `len(indexes) == len(queries)`
consider: `indexes = range(len(queries))`